### PR TITLE
Add public docs for device trust

### PR DESCRIPTION
**Pull request by codingllama**
Add the following public documentation:

* Device Trust guide under access control
* Device Trust configuration reference
* `tctl devices` command reference
* `tsh device enroll` command reference

https://github.com/gravitational/teleport.e/issues/514

### CODE REVIEW

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+trusted device, in addition to establishing the user's identity and enforcing
+the necessary roles. Furthermore, users using a trusted device leave audit
+trails that include the device's information.
+
+The device trust preview supports the following Teleport features:
+
+- SSH access enforcement
+- Database access enforcement
+- Kubernetes access enforcement
+- macOS trusted devices
```
Comment (zmb3) [inline docs/pages/access-controls/guides/device-trust.mdx:25]: The last one is more of a system requirement than a Teleport feature.

I would say "the device trust preview requires macOS and supports the following Teleport features:"

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+
+Support for other operating systems and access features is planned for upcoming
+Teleport versions.
+
+## Concepts
```
Comment (zmb3) [inline docs/pages/access-controls/guides/device-trust.mdx:30]: Some of the feedback we've seen from the community lately is a desire to see more "how it works" docs instead of framing everything as a guide.

I wonder if it makes sense to pull some of these concepts out into a separate intro page for people who want to understand the feature but don't care (or aren't ready) to read through a step-by-step guide on how to set it up.

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+  # unrelated fields omitted.
+spec:
+  roles:
+  - access
+  - device-admin # add this line
+version: v2
+```
+
+```code
+$ tctl edit users/codingllama
```
Comment (zmb3) [inline docs/pages/access-controls/guides/device-trust.mdx:132]: First use of `tctl edit` in the docs! 🎉

**Review by zmb3 — APPROVED** [review]

File: docs/pages/access-controls/guides/device-trust.mdx (hunk 3 above)
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:132]: `tctl edit` is too good. I wish we could `tctl edit` resources that don't exist yet...  ;)

File: docs/pages/access-controls/guides/device-trust.mdx (hunk 2 above)
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:30]: I thought briefly about a separate page too. Maybe we could move this part under architecture?

The reason I added it here is that, unlike other features, device trust isn't self-explanatory. Even a weakly-opinionated system like our has to make some calls on how devices are registered, enrolled, etc, so it felt necessary to explain a bit of how it works here. I'm open to a refactor, though.

File: docs/pages/access-controls/guides/device-trust.mdx (hunk 1 above)
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:25]: Rephrased as "The device trust preview works only with macOS devices and supports the following Teleport features". Let me know if you prefer your wording.

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+# user "codingllama" has been updated
+$ tsh logout; tsh login --proxy=example.com --user=codingllama ## fetch new certificates
+```
+
+Having the necessary permissions, let's add a device to the inventory. Replace
+the `$serial` below with your macOS device serial number (visible under Apple ->
+"About This Mac" -> "Serial number").
+
+```code
+$ serial="$(ioreg -c IOPlatformExpertDevice -d 2 | grep -i IOPlatformSerialNumber | awk -F'"' '{print $4}')"
```
Comment (alexfornuto) [inline docs/pages/access-controls/guides/device-trust.mdx:142]: Above we tell the reader to replace `$serial` (hence my earlier comment) but here we're explicitly defining it.

Should we make it an uppercase var to follow convention?

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+# C00AA3BBBB6C macOS not enrolled  d40f2ee4-856d-4aef-b784-c4371e39c036
+```
+
+A registered device is a device that Teleport knows about. Once a registered
+device is enrolled, it becomes a trusted device.
+
+<Admonition type="tip" title="Removing Devices">
+  A device that is no longer in use may be removed using
+  `tctl devices rm --device-id=<ID>` or
+  `tctl devices rm --asset-tag=<SERIAL_NUMBER>`.
```
Comment (alexfornuto) [inline docs/pages/access-controls/guides/device-trust.mdx:162]:
```suggestion
A device that is no longer in use may be removed using
`tctl devices rm --device-id=<ID>` or
`tctl devices rm --asset-tag=<SERIAL_NUMBER>`.
```

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+enrollment ceremony. Enrollment exchanges an enrollment token, created by a
+device admin, for the opportunity to enroll the corresponding device.
+
+To create an enrollment token run the following command, where `--asset-tag` is
+the serial number of the device we want to enroll.
+
+```code
+$ tctl devices enroll --asset-tag="C00AA3BBBB6C"
+# Run the command below on device "C00AA3BBBB6C" to enroll it:
+# tsh device enroll --token=MQi9gkOfyvdyNyN2moJZz9cqx/GyILR5uNu4NE9dxO8
```
Comment (alexfornuto) [inline docs/pages/access-controls/guides/device-trust.mdx:177]:
```suggestion
Run the command below on device "C00AA3BBBB6C" to enroll it:
tsh device enroll --token=MQi9gkOfyvdyNyN2moJZz9cqx/GyILR5uNu4NE9dxO8
```

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 7]
```diff
@@ -0,0 +1,326 @@
...
+```
+
+To perform the enrollment ceremony, using device specified above, type the
```
Comment (alexfornuto) [inline docs/pages/access-controls/guides/device-trust.mdx:180]:
```suggestion
To perform the enrollment ceremony, using the device specified above, type the
```

File: docs/pages/access-controls/guides/device-trust.mdx (hunk 6 above)
Comment (alexfornuto) [reply docs/pages/access-controls/guides/device-trust.mdx:177]: We should consider if we want to create a preset enrollment token in `config.json` that's obviously fake and use it here and anywhere else we end up talking about device enrollment.

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 8]
```diff
@@ -0,0 +1,326 @@
...
+$ tsh device enroll --token=MQi9gkOfyvdyNyN2moJZz9cqx/GyILR5uNu4NE9dxO8
+# Device "C00AA3BBBB6C"/macOS enrolled
+
+$ tsh logout; tsh login --proxy=example.com --user=codingllama # fetch new certificates
+```
+
+<Admonition type="warning" title="macOS">
+  macOS enrollments are tied to the current user in the device.
+  If the same device is handed to a new user, they will need to perform
+  enrollment again.
```
Comment (alexfornuto) [inline docs/pages/access-controls/guides/device-trust.mdx:193]:
```suggestion
macOS enrollments are tied to the current user in the device.
If the same device is handed to a new user, they will need to perform
enrollment again.
```

Comment (alexfornuto) [reply docs/pages/access-controls/guides/device-trust.mdx:193]: IMHO instead of a warning after the reader completes the process, this should be explained at the start of the section, before an admin following along enrolls a device to their own user.

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 9]
```diff
@@ -0,0 +1,326 @@
...
+    ```
+
+    Update the configuration:
+
+    ```code
+    $ tctl create -f cap.yaml
+    # cluster auth preference has been updated
+    ```
+  </TabItem>
+</Tabs>
```
Comment (alexfornuto) [inline docs/pages/access-controls/guides/device-trust.mdx:264]: Edits for copy, indents, and output formatting:

````suggestion
<Tabs>
<TabItem label="Static Config">
Edit the Auth Server's `teleport.yaml` file:

```yaml
auth_service:
  authentication:
    type: local
    second_factor: "on"
    webauthn:
      rp_id: example.com
    device_trust:
      mode: "required" # add this line
```

</TabItem>
<TabItem label="Dynamic resources">
Create a `cap.yaml` file or get the existing configuration using
`tctl get cluster_auth_preference`:

```yaml
kind: cluster_auth_preference
version: v2
metadata:
  name: cluster-auth-preference
spec:
  type: local
  second_factor: "on"
  webauthn:
    rp_id: example.com
  device_trust:
    mode: "required" # add this line
```

Update the configuration:

```code
$ tctl create -f cap.yaml
cluster auth preference has been updated
```

</TabItem>
</Tabs>
````

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 10]
```diff
@@ -0,0 +1,326 @@
...
+$ tctl create -f cap.yaml
+# cluster auth preference has been updated
+```
+</ScopedBlock>
+
+SSH, Database and Kubernetes access without a trusted device is now forbidden.
+For example, SSH access without a trusted device fails with the following error:
+
+```
+$ tsh ssh zarquon
```
Comment (alexfornuto) [inline docs/pages/access-controls/guides/device-trust.mdx:297]: Replacing with a more generic example name:

```suggestion
$ tsh ssh server1
```

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 11]
```diff
@@ -0,0 +1,326 @@
...
+# ERROR: ssh: rejected: administratively prohibited (unauthorized device)
+```
+
+<Admonition type="tip" title="Trusted Clusters">
+  It is possible to use [trusted
+  clusters](../../management/admin/trustedclusters.mdx) to limit the impact of
+  device mode `required`. A leaf cluster in mode `required` will enforce access
+  to all of its resources, without imposing the same restrictions to the root
+  cluster.
```
Comment (alexfornuto) [inline docs/pages/access-controls/guides/device-trust.mdx:306]:
```suggestion
It is possible to use [trusted
clusters](../../management/admin/trustedclusters.mdx) to limit the impact of
device mode `required`. A leaf cluster in mode `required` will enforce access
to all of its resources, without imposing the same restrictions to the root
cluster.
```

File: docs/pages/reference/cli.mdx [hunk 12]
```diff
@@ -1096,6 +1096,29 @@ $ tsh request drop [<request-id>...]
...
+
+| Name | Default Value(s) | Allowed Value(s) | Description |
+| - | - | - | - |
+| `--token` | none | String | Device enrollment token |
+
+#### Examples
+
+```code
+$ tsh device enroll --token=f3oJYqWfO6XG1ZCIF+faz8HIpDye54LsRFObftGKVVU
+# Device "C00AA3BBBB6C"/macOS enrolled
```
Comment (alexfornuto) [inline docs/pages/reference/cli.mdx:1119]: Another spot to replace with an obviously fake token. Maybe we should also create a dummy serial number var as well?

```suggestion
$ tsh device enroll --token=f3oJYqWfO6XG1ZCIF+faz8HIpDye54LsRFObftGKVVU
Device "C00AA3BBBB6C"/macOS enrolled
```

File: docs/pages/reference/cli.mdx [hunk 13]
```diff
@@ -2373,6 +2396,113 @@ $ tctl alerts create \
...
+
+```code
+$ tctl devices add --os=OS --asset-tag=SERIAL_NUMBER
+```
+
+#### Flags
+
+| Name | Default Value(s) | Allowed Value(s) | Description |
+| - | - | - | - |
+| `--os` | none | `linux`, `macos`, `windows` | Device operating system |
```
Comment (alexfornuto) [inline docs/pages/reference/cli.mdx:2412]: If we're listing all eventual values for `--os`, should we add a note to the reference that only `macos` is currently available?

File: docs/pages/reference/cli.mdx [hunk 14]
```diff
@@ -2373,6 +2396,113 @@ $ tctl alerts create \
     "The system is under maintenance, functionality may be limited."
 ```
 
+### tctl devices add
```
Comment (alexfornuto) [inline docs/pages/reference/cli.mdx:2399]: Are these entries being added in any sort of structured order? If not, may I suggest listing them alphabetically?

**Review by alexfornuto — COMMENTED** [review]
I reviewed in GitHub only (haven't seen the rendered version), and haven't yet tested.

File: docs/pages/access-controls/guides/device-trust.mdx (hunk 6 above)
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:177]: Enrollment tokens are random blobs, there's nothing special encoded on them. Happy to use a common token if you think it's best, though.

File: docs/pages/reference/cli.mdx (hunk 12 above)
Comment (codingllama) [reply docs/pages/reference/cli.mdx:1119]: That serial number is a dummy, but let me know if you'd rather make it more obvious.

File: docs/pages/reference/cli.mdx (hunk 13 above)
Comment (codingllama) [reply docs/pages/reference/cli.mdx:2412]: I skipped that on purpose here so there is less risk of forgetting to update it later. The guide does spell it out, as does the command if you try to use it today.

File: docs/pages/reference/cli.mdx (hunk 14 above)
Comment (codingllama) [reply docs/pages/reference/cli.mdx:2399]: The entries are sorted according to `tctl devices --help`, which uses a logical ordering of sorts: device verbs first (add,ls,rm), then enrollment tokens (enroll). Happy to change as well.

File: docs/pages/access-controls/guides/device-trust.mdx (hunk 6 above)
Comment (alexfornuto) [reply docs/pages/access-controls/guides/device-trust.mdx:177]: Only because we've been told int the past (can't find the issue/PR at the moment) that we get false-positive security reports for having example tokens that look real enough.

File: docs/pages/reference/cli.mdx (hunk 14 above)
Comment (alexfornuto) [reply docs/pages/reference/cli.mdx:2399]: Following that convention makes sense from that perspective. I'm thinking about the perspective of the user who doesn't have this level of headers listed in our ToC. They may know what they're looking for and be using ctrl/cmd+f to find it, or they may be scrolling the entire document/section to learn. Especially in the latter case, alphabetical ordering can help them keep from losing place.

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 15]
```diff
@@ -0,0 +1,326 @@
...
+token. A device enrollment token needs to be created by a device admin and sent
+to the person performing the enrollment ceremony. The token ties the actual
+device being enrolled to the expected device in the inventory, adding an extra
+layer of protection.
+
+## Prerequisites
+
+(!docs/pages/includes/commercial-prereqs-tabs.mdx!)
+
+- A signed and notarized `tsh` binary for enrollment.
```
[the remark this answers is not in the document]
Comment (zmb3) [reply docs/pages/access-controls/guides/device-trust.mdx:67]: They do need a special version of the binary, because we still ship both signed and unsigned mac binaries (unfortunately).

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 16]
```diff
@@ -0,0 +1,326 @@
+---
+title: "Device Trust (Preview)"
+description: Learn how to use and enforce trusted devices with Teleport
+---
+
+<Admonition type="warning" title="Preview">
+  Device Trust is currently in Preview mode.
```
[the remark this answers is not in the document]
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:7]: Done.

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 17]
```diff
@@ -0,0 +1,326 @@
...
+</Admonition>
+
+<ScopedBlock scope="oss">
+  Device Trust requires a commercial edition of Teleport.
```
[the remark this answers is not in the document]
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:11]: 1. Done. I was mostly following other <Admonition> tags.
2. Done. Ditto, this follows the wording from <Admonitions> in other pages.

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 18]
```diff
@@ -0,0 +1,326 @@
...
+- Database access enforcement
+- Kubernetes access enforcement
+
+Support for other operating systems and access features is planned for upcoming
+Teleport versions.
+
+## Concepts
+
+Device Trust uses the Secure Enclave in macOS devices to store a device private
+key. That key is used to solve device challenges, issued by the Teleport Auth
```
[the remark this answers is not in the document]
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:33]: done.

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 19]
```diff
@@ -0,0 +1,326 @@
...
+Server, thus proving the identity of the trusted device.
```
[the remark this answers is not in the document]
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:34]: Done.

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 20]
```diff
@@ -0,0 +1,326 @@
...
+
+Device management is divided into two separate phases: inventory management and
+device enrollment.
+
+__Inventory management__ is performed by a device admin. In this step, devices
```
[the remark this answers is not in the document]
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:39]: Done throughout.

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 21]
```diff
@@ -0,0 +1,326 @@
...
+at your discretion. This is the step that creates the Secure Enclave private key
+in the device and register its public key counterpart with the Teleport Auth
+Server. Enrollment has to run in the actual device that you want to enroll. For
+example, this happens when a user gets a new device for the first time, or when
+IT prepares a new device for a user. Enrollment only needs to happen once per
+user/device combination.
+
+<Admonition type="tip" title="Enrollment">
+  Device enrollment is key in correctly identifying a device throughout its
+  lifetime. The more trusted the enrollment operator, the better are the ongoing
```
[the remark this answers is not in the document]
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:53]: Done. Let me know if you have other suggestions to make it better.

File: docs/pages/access-controls/guides/device-trust.mdx (hunk 15 above)
[the remark this answers is not in the document]
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:67]: Adding to what Zac said, even the "signed" standalone `tsh` we ship in the tar.gz won't do, as it needs to be in the "tsh.app" shell to work properly. #14006 is the issue meant to address this.

In regards to `tctl`, the include above does cover it, that's why it isn't listed again.

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 22]
```diff
@@ -0,0 +1,326 @@
...
+      - list
+      - update
+      - delete
+      - create_enroll_token
+      - enroll
+```
+
+```code
+$ tctl create -f device-admin.yaml
+# role 'device-admin' has been created
```
[the remark this answers is not in the document]
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:102]: Done for all code blocks in this file.

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 23]
```diff
@@ -0,0 +1,326 @@
...
+```
+
+<Admonition type="tip" title="Editor Role">
+  For clusters created using Teleport v12 or newer, the builtin role `editor`
+  has the necessary permissions to manage devices.
+  For production setups a separate role is still recommended.
+</Admonition>
+
+Note that, in addition to the usual CRUD verbs (create, read, list, update and
```
[the remark this answers is not in the document]
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:111]: Done.

File: docs/pages/access-controls/guides/device-trust.mdx (hunk 7 above)
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:180]: Done.

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 24]
```diff
@@ -0,0 +1,326 @@
...
+  # unrelated fields omitted.
+spec:
+  roles:
+  - access
+  - device-admin # add this line
+version: v2
+```
+
+```code
+$ tctl edit users/codingllama
```
[the remark this answers is not in the document]
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:132]: Done in a way - I show the `tctl edit` then the diff block, but skipped the `tctl edit` output, as it looks odd to repeat the command.

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 25]
```diff
@@ -0,0 +1,326 @@
...
+
+Note that, in addition to the usual CRUD verbs (create, read, list, update and
+delete), we have also included `create_enroll_token` and `enroll`. The
+`create_enroll_token` verb is necessary to execute the `tctl devices enroll`
+command; `enroll` is necessary to execute `tsh device enroll`. Both are
+demonstrated in the following section.
+
+Make sure your user has the `device-admin` role:
+
+```yaml
```
[the remark this answers is not in the document]
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:119]: Good call, done here and for others.

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 26]
```diff
@@ -0,0 +1,326 @@
...
+kind: user
+metadata:
+  name: codingllama
```
[the remark this answers is not in the document]
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:122]: Done for all.

File: docs/pages/access-controls/guides/device-trust.mdx (hunk 10 above)
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:297]: Done.

File: docs/pages/access-controls/guides/device-trust.mdx (hunk 4 above)
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:142]: Yeah, I didn't like this one too much, but it's important to explain what the asset tag is and the command is kinda hard to parse.

I've added a nod to the script in the parenthesis above.

> Should we make it an uppercase var to follow convention?

Done. Do we have our conventions documented somewhere? Browsing other docs for examples is often proving unreliable to me.

File: docs/pages/access-controls/guides/device-trust.mdx (hunk 8 above)
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:193]: Makes sense, moved.

File: docs/pages/access-controls/guides/device-trust.mdx (hunk 9 above)
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:264]: Done.

File: docs/pages/reference/cli.mdx (hunk 14 above)
Comment (codingllama) [reply docs/pages/reference/cli.mdx:2399]: I find this page a bit confusing too, with the lack of navigation and the fact that most of it seems to have no sensible order. Maybe we could add navigation and/or sort the commands consistently throughout? (In a separate PR, that is.)

Sorted `tctl devices` commands alphabetically, but left everything else as-is.

File: docs/pages/access-controls/guides/device-trust.mdx (hunk 6 above)
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:177]: Done for asset tag and enroll tokens, all files/occurrences.

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 27]
```diff
@@ -0,0 +1,326 @@
...
+```
+
+```code
+$ tctl edit users/codingllama
+# user "codingllama" has been updated
+$ tsh logout; tsh login --proxy=example.com --user=codingllama ## fetch new certificates
+```
+
+Having the necessary permissions, let's add a device to the inventory. Replace
+the `$serial` below with your macOS device serial number (visible under Apple ->
```
[the remark this answers is not in the document]
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx]: Sorry, I'm not sure what you mean here.

[the remark this answers is not in the document]
Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx]: I gave this a try, but unless I ditch the helper commands below the end result seems worse than now.

**Comment by codingllama** [comment]
Friendly ping @alexfornuto ?

**Review by zmb3 — APPROVED** [review]

**Comment by alexfornuto** [comment]
Pings received, will review again today.

File: docs/pages/access-controls/guides/device-trust.mdx (hunk 2 above)
Comment (alexfornuto) [reply docs/pages/access-controls/guides/device-trust.mdx:30]: I think the volume of explanation here is small enough to warrant a single page. Any more, and we'd want to separate it.

File: docs/pages/access-controls/guides/device-trust.mdx [hunk 28]
```diff
@@ -0,0 +1,298 @@
...
+$ tsh logout; tsh login --proxy=example.com --user=alice
+```
+
+Having the necessary permissions, let's add a device to the inventory. Replace
+the `$SERIAL` below with your macOS device serial number (visible under Apple ->
+"About This Mac" -> "Serial number", or use the command below).
+
+```code
+$ SERIAL="$(ioreg -c IOPlatformExpertDevice -d 2 | grep -i IOPlatformSerialNumber | awk -F'"' '{print $4}')"
+$ tctl devices add --os=macos --asset-tag="$SERIAL"
```
Comment (alexfornuto) [inline docs/pages/access-controls/guides/device-trust.mdx:147]: It was at this point that my testing failed:

```
❯ tsh version
Teleport v12.0.0-alpha.2 git:api/v12.0.0-alpha.2-0-g3ebfce4a6a go1.19.5
Proxy version: 12.0.0-alpha.2

❯ tctl version
Teleport v12.0.0-alpha.2 git:api/v12.0.0-alpha.2-0-g3ebfce4a6a go1.19.5

❯ tctl get users/alex
kind: user
metadata:
  ...
  name: alex
spec:
  ...
  roles:
  - editor
  - access
  - device-admin
  status:
   ...
version: v2

❯ tctl get roles/device-admin
kind: role
metadata:
  ...
  name: device-admin
spec:
  allow:
    ...
    rules:
    - resources:
      - device
      verbs:
      - create
      - read
      - list
      - update
      - delete
      - create_enroll_token
      - enroll
  deny: {}
  options:
    ...
version: v5

❯ echo $SERIAL
REDACTED

❯ tctl devices add --os=macos --asset-tag="$SERIAL"
tctl: error: expected command but got "devices"

```

Comment (codingllama) [reply docs/pages/access-controls/guides/device-trust.mdx:147]: You'll need the Enterprise version of `tctl` to do this part. We can make that clearer in the docs, or add to the troubleshooting section. WDYT?

Alternatively, we could move this code to OSS `tctl`, say for v12.1. @zmb3 for an opinion.

File: docs/pages/reference/cli.mdx (hunk 14 above)
Comment (alexfornuto) [reply docs/pages/reference/cli.mdx:2399]: Thanks! I spun off an issue to tackle the larger doc's sorting.

File: docs/pages/access-controls/guides/device-trust.mdx (hunk 28 above)
Comment (alexfornuto) [reply docs/pages/access-controls/guides/device-trust.mdx:145]: I think we can make this more obvious by updating the `/commercial-prereqs-tabs.mdx` codeblock to show the Enterprise output of `tctl version`.

**Review by alexfornuto — APPROVED** [review]
Tested and confirmed!

```
❯ tsh ssh ubuntu@ip-10-0-1-16
ubuntu@ip-10-0-1-16:~$
```

```
alex@Xeon:~$ tsh ssh ubuntu@ip-10-0-1-16
ERROR: ssh: rejected: administratively prohibited (unauthorized device)
```

I made some minor updates, commit messages should be verbose enough to explain them.

File: docs/pages/access-controls/guides/device-trust.mdx (hunk 4 above)
Comment (alexfornuto) [reply docs/pages/access-controls/guides/device-trust.mdx]: The convention I'm referring to isn't our own.

[Reference 1](https://www.tutorialspoint.com/unix/unix-using-variables.htm#:~:text=The%20name%20of%20a%20variable,have%20their%20names%20in%20UPPERCASE.)

[Reference 2](https://stackoverflow.com/questions/673055/correct-bash-and-shell-script-variable-capitalization)

**Comment by codingllama** [comment]
Thanks for the reviews and contributions! Merging.
